From eeecf8a399080598e5290d3356b0ad557bd0ccbd Mon Sep 17 00:00:00 2001
From: Alin Jerpelea <alin.jerpelea@sony.com>
Date: Thu, 9 Aug 2018 09:34:06 +0200
Subject: [PATCH] hal: enable audio hal on sdm660

the sdm660 SOC can use the same HAL as the other QCOM SOCs

Test: manual - build, boots and audio works

Change-Id: I0edd5fa2c67eb7a96a44e907060dcbb273e983ac
Signed-off-by: Alin Jerpelea <alin.jerpelea@sony.com>

post_proc: Enable post processing for sdm660

the sdm660 SOC can use the same prost_proc code as all the other
QCOM SOCs

Test: manual - builds, boots, sound works

Change-Id: I7bb807788e457f7ec6ce5124dfb1d88dc96d8127
Signed-off-by: Alin Jerpelea <alin.jerpelea@sony.com>
Signed-off-by: Tim <tomgus1.2@gmail.com>
---
 Android.mk            | 2 +-
 hal/Android.mk        | 2 +-
 hal/msm8916/hw_info.c | 7 ++++++-
 post_proc/Android.mk  | 4 ++--
 4 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/Android.mk b/Android.mk
index 41109b0..e23010d 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,6 +1,6 @@
 # TODO:  Find a better way to separate build configs for ADP vs non-ADP devices
 ifneq ($(TARGET_BOARD_AUTO),true)
-  ifneq ($(filter msm8960 msm8226 msm8x26 msm8x84 msm8084 msm8992 msm8994 msm8996 msm8909 msm8952 msm8998 sdm845 sdm710 sm8150,$(TARGET_BOARD_PLATFORM)),)
+  ifneq ($(filter msm8960 msm8226 msm8x26 msm8x84 msm8084 msm8992 msm8994 msm8996 msm8909 msm8952 msm8998 sdm660 sdm845 sdm710 sm8150,$(TARGET_BOARD_PLATFORM)),)
 
     MY_LOCAL_PATH := $(call my-dir)
 
diff --git a/hal/Android.mk b/hal/Android.mk
index 05455b5..cb8f8af 100644
--- a/hal/Android.mk
+++ b/hal/Android.mk
@@ -71,7 +71,7 @@ ifneq ($(filter sm8150,$(TARGET_BOARD_PLATFORM)),)
 endif
 endif
 
-ifneq ($(filter msm8916 msm8909 msm8952,$(TARGET_BOARD_PLATFORM)),)
+ifneq ($(filter msm8916 msm8909 msm8952 sdm660,$(TARGET_BOARD_PLATFORM)),)
   AUDIO_PLATFORM = msm8916
   LOCAL_CFLAGS := -DPLATFORM_MSM8916
 ifneq ($(filter msm8909,$(TARGET_BOARD_PLATFORM)),)
diff --git a/hal/msm8916/hw_info.c b/hal/msm8916/hw_info.c
index 095d052..de4e6a7 100644
--- a/hal/msm8916/hw_info.c
+++ b/hal/msm8916/hw_info.c
@@ -56,8 +56,12 @@ static void update_hardware_info_8x16(struct hardware_info *hw_info, const char
         strlcpy(hw_info->name, "msm8952", sizeof(hw_info->name));
     } else if (!strcmp(snd_card_name, "msm8976-tasha-snd-card")) {
         strlcpy(hw_info->name, "msm8976", sizeof(hw_info->name));
+    } else if (!strcmp(snd_card_name, "sdm660-snd-card")) {
+        strlcpy(hw_info->name, "sdm660", sizeof(hw_info->name));
+    } else if (!strcmp(snd_card_name, "sdm660-tasha-snd-card")) {
+        strlcpy(hw_info->name, "sdm660", sizeof(hw_info->name));
     } else {
-        ALOGW("%s: Not an 8x16/8909/8952/8976 device", __func__);
+        ALOGW("%s: Not an 8x16/8909/8952/8976/sdm660 device", __func__);
     }
 }
 
@@ -73,6 +77,7 @@ void *hw_info_init(const char *snd_card_name)
 
     if (strstr(snd_card_name, "msm8x16") || strstr(snd_card_name, "msm8909") ||
         strstr(snd_card_name, "msm8952") || strstr(snd_card_name, "msm8976") ||
+	strstr(snd_card_name, "sdm660") ||
         strstr(snd_card_name, "msm-bg-snd-card")) {
         ALOGV("8x16 - variant soundcard");
 
diff --git a/post_proc/Android.mk b/post_proc/Android.mk
index c800d8f..5677f81 100644
--- a/post_proc/Android.mk
+++ b/post_proc/Android.mk
@@ -1,4 +1,4 @@
-ifneq ($(filter msm8974 msm8226 msm8084 msm8992 msm8994 msm8996 msm8909 msm8952 msm8998 sdm845 sdm710 sm8150,$(TARGET_BOARD_PLATFORM)),)
+ifneq ($(filter msm8974 msm8226 msm8084 msm8992 msm8994 msm8996 msm8909 msm8952 msm8998 sdm660 sdm845 sdm710 sm8150,$(TARGET_BOARD_PLATFORM)),)
 
 LOCAL_PATH:= $(call my-dir)
 
@@ -43,7 +43,7 @@ endif
 
 ################################################################################
 
-ifneq ($(filter msm8992 msm8994 msm8996 msm8909 msm8952 msm8998 sdm845 sdm710 sm8150,$(TARGET_BOARD_PLATFORM)),)
+ifneq ($(filter msm8992 msm8994 msm8996 msm8909 msm8952 msm8998 sdm660 sdm845 sdm710 sm8150,$(TARGET_BOARD_PLATFORM)),)
 
 include $(CLEAR_VARS)
 
-- 
2.13.0

